docs(animations): add links to `state()` references

dario-piotrowicz opened the pull request:
the keyword 'state' is included in the `ignoredWords` set that prevents
certain words to be autolinked, this causes the animations' state
function not to be automatically linked, so manually link those
references to the state api docs

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## Issue

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

dario-piotrowicz committed:
docs(animations): add links to `state()` references

the keyword 'state' is included in the `ignoredWords` set that prevents
certain words to be autolinked, this causes the animations' state
function not to be automatically linked, so manually link those
references to the state api docs

dario-piotrowicz commented on aio/content/guide/animations.md:
```diff
@@ -82,11 +82,11 @@ In HTML, these attributes are set using ordinary CSS styles such as color and op
 ### Animation state and styles
 
-Use Angular's `state()` function to define different states to call at the end of each transition. This function takes two arguments: a unique name like `open` or `closed` and a `style()` function.
+Use Angular's <code>[state](api/animations/state)()</code> function to define different states to call at the end of each transition. This function takes two arguments: a unique name like `open` or `closed` and a `style()` function.
```
@gkalpak the ugly/verbose version that make it look exactly like the auto-linked ones in the website :grimacing: 

![Screenshot at 2021-12-04 13-57-36](https://user-images.githubusercontent.com/61631103/144712182-e3a9658d-23c8-4331-b35c-3e82d8a48646.png)

dario-piotrowicz commented on packages/animations/src/animation_metadata.ts:
```diff
@@ -98,7 +98,7 @@ export declare interface AnimateChildOptions extends AnimationOptions {
   /**
    * Associates a named animation state with a set of CSS styles.
-   * See `state()`
+   * See [`state()`](api/animations/state)
```
@gkalpak in the ts files I went with this version and not the more verbose one as these files can be also viewed as simple source code so I thought that adding that would be a bit too much here

but this causes the parenthesis to be include in the link :sweat: 
![Screenshot at 2021-12-04 13-54-49](https://user-images.githubusercontent.com/61631103/144712141-1ce9ec3e-356e-4284-9c6c-b46d6317e7bb.png)

What do you think?

gkalpak reviewed (approved):
Thx, @dario-piotrowicz!

Reviewed-for: global-docs-approvers

gkalpak commented:
**merge-assistance:** These are docs-only changes and shouldn't need a presubmit.

dario-piotrowicz commented:
@gkalpak sorry I just noticed I missed some instances of `state` :bow: , just added them as a fixup commit, I hope it's fine :slightly_smiling_face:

dario-piotrowicz commented on aio/content/guide/animations.md:
```diff
@@ -154,13 +154,13 @@ region="transition2">
 <div class="alert is-helpful">
 
-**Note:** Some additional notes on using styles within `state` and `transition` functions.
+**Note:** Some additional notes on using styles within [`state`](api/animations/state) and `transition` functions.
```
@gkalpak unfortunately here we get to slightly different styles :sweat: :
![Screenshot at 2021-12-07 18-56-01](https://user-images.githubusercontent.com/61631103/145089458-f1378073-9c4b-4a2a-a136-c76a8c7158d4.png)

what do you think? should I just add the brackets so that they will align?

gkalpak replied:
Yes, please 🙏 
(Or maybe we could look into making the styles the same.)

dario-piotrowicz replied:
let me have a look :slightly_smiling_face: 

I feel like making the styles the same could be tricky (depending on how global this is) and proabaly should be done in a separate PR :thinking: 

(PS: sorry I made you just and remove the merge label at the last minute :sweat_smile: :bow: :pray: )

dario-piotrowicz replied:
@gkalpak `code` elements inside `a` tags were missing a padding-top and padding-bottom of 4px (non-auto linked code links are a > code whilst the auto-linked are code > a), I've added it and all looks ok now, I think this is totally fine, but it the `:not(a) > code` was purposely applied there...

can you see/think of any issue in adding the padding? :thinking:

mary-poppins commented:
You can preview df90a24 at https://pr44376-df90a24.ngbuilds.io/.
You can preview 30b9ca6 at https://pr44376-30b9ca6.ngbuilds.io/.
You can preview a6a1f75 at https://pr44376-a6a1f75.ngbuilds.io/.

gkalpak reviewed (changes requested):
Thx for making the changes :+1:

dario-piotrowicz commented on packages/animations/src/animation_metadata.ts:
```diff
@@ -450,12 +450,13 @@ export interface AnimationStaggerMetadata extends AnimationMetadata {
  * @param name An identifying string.
- * @param definitions  An animation definition object, containing an array of `state()`
+ * @param definitions  An animation definition object, containing an array of
+ [`state()`](api/animations/state)
  * and `transition()` declarations.
```
sorry :sweat::+1:

dario-piotrowicz commented on aio/src/styles/2-modules/code/_code.scss:
```diff
@@ -121,10 +121,6 @@ aio-code {
-    }
-
-    &:not(a) > code {
       padding: 4px;
```
did you maybe mean this with your suggestion?
![Screenshot at 2021-12-09 22-12-42](https://user-images.githubusercontent.com/61631103/145484027-7eca2bd0-fdbc-4bb3-9106-dee878ee30ac.png)

because this does seem to do the trick but as we've already discussed in some previous PRs, `is` isn't unfortunately suppored in IE11 :sweat: (https://caniuse.com/css-matches-pseudo)

(which I think you told me we still would want to support in aio?)

gkalpak replied:
Ah, you are right, I meant something like `:is()`.
Actually, since Angular v13 has dropped IE, we can now drop support for it in AIO as well.

dario-piotrowicz replied:
ah ok :smile: 

ok I went with your solution using the `line-height` mixin :slightly_smiling_face:  (as it looks to me more robust compared to `initial` :slightly_smiling_face: )

it's looking ok, I the link in the [workspace-config](https://pr44376-a6a1f75.ngbuilds.io/guide/workspace-config#generation-schematics) guide and there is like a small different between this new version and what is currently live in aio, something like 1 or 2 pixels, barely noticeable

please have a look and let me know what you think :slightly_smiling_face: 

(also, I figured this would sit nicely as it's own commit to keep things clearer, I hope it's fine, if it's not please let me know and I can merge all the changes in a single commit)

gkalpak replied:
> ok I went with your solution using the `line-height` mixin 🙂 (as it looks to me more robust compared to `initial` 🙂)

Looking at it again, `line-height: initial` seems to make the anchor-wrapped code elements the same height as the plain code elements (e.g. in [guide/workspace-config#generation-schematics][1], compare `@schematics/angular` with `ng generate` on the next line). But it is a small difference and I am fine either way :smiley:

So, overall, looks great to me 🚀 

[1]: https://pr44376-07f3ac0.ngbuilds.io/guide/workspace-config#generation-schematics

dario-piotrowicz replied:
if you prefer I can go with the `initial` I don't mind that much, I just prefer the mixin as it is more predictable I believe, but if you want I am ok with `initial` too :slightly_smiling_face:

gkalpak replied:
As I said, I am fine either way, so happy to go with your preference (since you put in all the work 😛).

mary-poppins commented:
You can preview 7630a1c at https://pr44376-7630a1c.ngbuilds.io/.

dario-piotrowicz committed:
fix(docs-infra): adjust code link styling height

the height of code links (anchors inside code tags or
code tags inside anchors) is inconsistent, fix such
issue by also making sure that code links to external
resources do not get wrongly styled

mary-poppins commented:
You can preview 07f3ac0 at https://pr44376-07f3ac0.ngbuilds.io/.

gkalpak reviewed (approved):
Thx, @dario-piotrowicz 💯

gkalpak commented:
**merge-assistance:** This is a docs-only change and should not need a g3 presubmit.

dylhunn commented:
This PR was merged into the repository by commit a864d1c926f86a88f5797b132733b828e24877e9.
